File: android/app/src/main/java/com/mylife/app/data/Record.kt
package com.mylife.app.data

import androidx.room.*
import java.util.Date

@Entity(tableName = "records")
data class Record(
    @PrimaryKey val id: String,
    val store: String,       // clothing, food, housing, transport
    val name: String,
    val category: String = "",
    val season: String = "",     // clothing only
    val calories: String = "",  // food only
    val cost: String = "",
    val note: String = "",
    val photo: String = "",     // base64 or file path
    val createdAt: Date = Date(),
    val updatedAt: Date = Date(),
)

@Dao
interface RecordDao {
    @Query("SELECT * FROM records WHERE store = :store ORDER BY createdAt DESC")
    suspend fun getByStore(store: String): List<Record>

    @Query("SELECT * FROM records WHERE id = :id")
    suspend fun getById(id: String): Record?

    @Query("SELECT COUNT(*) FROM records WHERE store = :store")
    suspend fun countByStore(store: String): Int

    @Query("SELECT * FROM records WHERE updatedAt > :since ORDER BY updatedAt ASC")
    suspend fun getUpdatedSince(since: Date): List<Record>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun upsert(record: Record)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun upsertAll(records: List<Record>)

    @Delete
    suspend fun delete(record: Record)

    @Query("DELETE FROM records WHERE id = :id AND store = :store")
    suspend fun deleteById(id: String, store: String)
}

File: backend/models/record.go
package models

import "time"

type Record struct {
	ID        string    `json:"id"`
	UserID    int64     `json:"user_id"`
	Store     string    `json:"store"` // clothing, food, housing, transport
	Name      string    `json:"name"`
	Category  string    `json:"category,omitempty"`
	Season    string    `json:"season,omitempty"`    // clothing only
	Calories  string    `json:"calories,omitempty"`  // food only
	Cost      string    `json:"cost,omitempty"`
	Note      string    `json:"note,omitempty"`
	Photo     string    `json:"photo,omitempty"`     // base64 data URL
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

var ValidStores = map[string]bool{
	"clothing":  true,
	"food":      true,
	"housing":   true,
	"transport": true,
}

type SyncPayload struct {
	Records []Record `json:"records"`
}

type StatsResponse struct {
	TotalRecords   int                `json:"total_records"`
	TotalCost      float64            `json:"total_cost"`
	ByStore        map[string]StoreStat `json:"by_store"`
}

type StoreStat struct {
	Count int     `json:"count"`
	Cost  float64 `json:"cost"`
}
